python/core/qgsvectorlayer.sip

```diff
@@ -1,306 +1,320 @@
+
+   Defines a character that is used as a decimal point in the numeric columns
+   The default is '.'.
+
+ - xField=column yField=column
+
+   Defines the name of the columns holding the x and y coordinates for XY point geometries.
+   If the useHeader is no (ie there are no column names), then this is the column
+   number (with the first column as 1).
+
+ - xyDms=(yes|no)
+
+   If yes then the X and Y coordinates are interpreted as
+   degrees/minutes/seconds format (fairly permissively),
+   or degree/minutes format.
+
+ - wktField=column
+
+   Defines the name of the columns holding the WKT geometry definition for WKT geometries.
+   If the useHeader is no (ie there are no column names), then this is the column
+   number (with the first column as 1).
+
+ - geomType=(point|line|polygon|none)
+
+   Defines the geometry type for WKT type geometries.  QGIS will only display one
+   type of geometry for the layer - any others will be ignored when the file is
+   loaded.  By default the provider uses the type of the first geometry in the file.
+   Use geomType to override this type.
+
+   geomType can also be set to none, in which case the layer is loaded without
+   geometries.
+
+ - subset=expression
+
+   Defines an expression that will identify a subset of records to display
+
+ - crs=crsstring
+
+   Defines the coordinate reference system used for the layer.  This can be
+   any string accepted by QgsCoordinateReferenceSystem::createFromString()
+
+ -subsetIndex=(yes|no)
+
+   Determines whether the provider generates an index to improve the efficiency
+   of subsets.  The default is yes
+
+ -spatialIndex=(yes|no)
+
+   Determines whether the provider generates a spatial index.  The default is no.
+
+ -watchFile=(yes|no)
+
+   Defines whether the file will be monitored for changes. The default is
+   to monitor for changes.
+
+ - quiet
+
+   Errors encountered loading the file will not be reported in a user dialog if
+   quiet is included (They will still be shown in the output log).
+
+ \subsection gpx GPX data provider (gpx)
+
+ Provider reads tracks, routes, and waypoints from a GPX file.  The url
+ defines the name of the file, and the type of data to retrieve from it
+ ("track", "route", or "waypoint").
+
+ An example url is "/home/user/data/holiday.gpx?type=route"
+
+ \subsection grass Grass data provider (grass)
+
+ Provider to display vector data in a GRASS GIS layer.
+
+ TODO QGIS3: Remove virtual from non-inherited methods (like isModified)
+ @see QgsVectorLayerUtils()
+/
+
+%End
+
 
   public:
 
     //! Result of an edit operation
     enum EditResult
     {
-      Success, /**< Edit operation was successful */
-      EmptyGeometry, /**< Edit operation resulted in an empty geometry */
-      EditFailed, /**< Edit operation failed */
-      FetchFeatureFailed, /**< Unable to fetch requested feature */
-      InvalidLayer, /**< Edit failed due to invalid layer */
+      Success = 0, //!< Edit operation was successful
```

Comment:
can we keep the enum values in SIP? I thought not but could not quickly find any reference. Do you know?